src/python/pants/rules/core/targets.py

```diff
@@ -166,12 +166,6 @@ def compute_value(
         return cast(Type[TargetV1], super().compute_value(raw_value, address=address))
 
 
-# TODO: maybe add a DictField? This could be a can of worms. How should we type the values? It
-#  doesn't seem safe to assume `Dict[str, str]`. We would want to use `FrozenDict` to store the
-#  computed value, but how should we handle converting the values into hashable types, e.g.
-#  converting a List[str] to Tuple[str, ...]?
-#
-#  Audit other uses of dict fields before adding something.
```

Comment:
We still end up using `PrimitiveField` here because the value is `Any`, rather than `str` or `Iterable[str]` :/ this field is weird and may need future attention. I'm not sure anyone is even using it, though.